File: tps/TP04/ArrayHash.java
import java.util.Scanner;

public class ArrayHash {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int casos = scanner.nextInt();
        scanner.nextLine(); // Consumir a quebra de linha após o número de casos

        while (casos > 0) {
            int qtsLinhas = scanner.nextInt();
            scanner.nextLine(); // Consumir a quebra de linha após o número de linhas

            int qtsCaso = 0;
            int somaLetras = 0;

            while (qtsLinhas > 0) {
                String string = scanner.nextLine();

                int i = 0;
                int qtsLetra = 0;

                while (i < string.length()) {
                    somaLetras += (string.charAt(i) - 'A') + qtsCaso + qtsLetra;

                    i++;
                    qtsLetra++;
                }

                qtsCaso++;

                qtsLinhas--;
            }

            System.out.println(somaLetras);

            casos--;
        }

        scanner.close();
    }
}

File: tps/TP04/ArrayHash.c
#include <stdio.h>
#include <string.h>

int main() {
    int casos, qtsLinhas;
    int i, qtsCaso;
    int qtsLetra;
    int somaLetras;

    char string[60];

    scanf("%d", &casos);

    while (casos--) {
        scanf("%d", &qtsLinhas);

        qtsCaso = 0;
        somaLetras = 0;
        while (qtsLinhas--) {
            scanf(" %s", string);

            i = 0;
            qtsLetra = 0;

            while (string[i]) {
                somaLetras += (string[i] - 'A') + qtsCaso + qtsLetra++;

                i++;
            }

            qtsCaso++;

            memset(string, 0, sizeof(string));
        }

        printf("%d\n", somaLetras);
    }

    return 0;
}
